File: Daily Challenge/Java/Sort Words - Embedded Integers 21.1.2026.java
import java.util.*;
public class Main {
    public static int getNum(String w){
        String num = "";
        for(char c : w.toCharArray())
            if(Character.isDigit(c)) num += c;
        return Integer.parseInt(num);
    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        String[] words = sc.nextLine().split(" ");
        Arrays.sort(words, (a, b) -> getNum(a) - getNum(b));
        for(String w : words) System.out.print(w + " ");
    }
}

File: Daily Challenge/C++/Sort Words - Embedded Integers 21.1.2026.cpp
#include <bits/stdc++.h>
using namespace std;

int get_num(const string &w) {
    string num;
    for (char c : w) if (isdigit(c)) num += c;
    return stoi(num);
}

int main() {
    string line;
    getline(cin, line);
    stringstream ss(line);
    vector<string> words;
    string w;
    while (ss >> w) words.push_back(w);

    sort(words.begin(), words.end(), [](const string &a, const string &b){
        return get_num(a) < get_num(b);
    });

    for (auto &word : words) cout << word << " ";
    cout << "\n";
}